src/GitHub.InlineReviews/Services/PullRequestSessionService.cs

```diff
@@ -592,6 +594,8 @@ public ITextDocument GetDocument(ITextBuffer buffer)
                 CommitId = result.CommitId,
                 DiffHunk = result.DiffHunk,
                 Id = result.Id,
+                //TODO: Populate this
+                NodeId = string.Empty,
```

Comment:
Could you elaborate on why this needs to be created and why the TODO comment (I'm totally clueless on this part of the code)

Reply:
I think this is where Stanley got stuck. The problem is that we're currently mixing GraphQL and REST queries and each of them has a different concept of ID, so we currently need both. #1712 will fix this properly by moving to GQL for all this stuff, but in the interim I think we can convert the two "Add Comment" methods that currently use REST, to using GQL; though will be a bit messy because there's currently no way to actually do this in a single GQL mutation.

Reply:
Apologies for the lack of communication here. I left my status with this in Slack.

@grokys is correct. I migrated some of his code from #1712 in order to fill the gaps.